core/src/main/java/brooklyn/entity/rebind/RebindIteration.java

```diff
@@ -341,38 +340,67 @@ protected void rebuildCatalog() {
                 }
             }
         }
+
+        // See notes in CatalogInitialization
         
-        // Register catalogue items with the management context. Loads the bundles in the OSGi framework.
-        CatalogLoadMode catalogLoadMode = managementContext.getConfig().getConfig(BrooklynServerConfig.CATALOG_LOAD_MODE);
+        Collection<CatalogItem<?, ?>> catalogItems = rebindContext.getCatalogItems();
+        CatalogInitialization catInit = ((ManagementContextInternal)managementContext).getCatalogInitialization();
+        catInit.injectManagementContext(managementContext);
```

Comment:
Why would we have to inject the management context here, when we've obtained the instance by calling  `managementContext.getCatalogInitialization()`? That should return an instance which already has the management context.